### PR TITLE
CRAFTERCMS-2593

### PR DESCRIPTION
**Pull request by dejan-brkic**
Implement cluster API

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/2593

### CODE REVIEW

File: src/main/java/org/craftercms/studio/api/v2/service/cluster/internal/ClusterManagementServiceInternal.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+package org.craftercms.studio.api.v2.service.cluster.internal;
+
+import org.craftercms.studio.api.v2.dal.ClusterMember;
+
+import java.util.List;
+
+public interface ClusterManagementServiceInternal {
+
+    List<ClusterMember> getAllMembers();
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v2/service/cluster/internal/ClusterManagementServiceInternal.java:32]: Java doc this file, please.

File: src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    @Override
+    public boolean addMember(ClusterMember clusterMember) {
+        if (StringUtils.isEmpty(clusterMember.getClusterId())) {
+            clusterMember.setClusterId(UUID.randomUUID().toString());
+        }
+        if (StringUtils.isEmpty(clusterMember.getClusterMemberName())) {
+            clusterMember.setClusterMemberName(UUID.randomUUID().toString());
+        }
+        if (StringUtils.isEmpty(clusterMember.getClusterMemberIp())) {
+            clusterMember.setClusterMemberIp("127.0.0.1");
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java]: Why?
Shouldn't this cause an error?

File: src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    @Override
+    public boolean updateMember(ClusterMember clusterMember) {
+        if (StringUtils.isEmpty(clusterMember.getClusterId())) {
+            clusterMember.setClusterId(UUID.randomUUID().toString());
+        }
+        if (StringUtils.isEmpty(clusterMember.getClusterMemberName())) {
+            clusterMember.setClusterMemberName(UUID.randomUUID().toString());
+        }
+        if (StringUtils.isEmpty(clusterMember.getClusterMemberIp())) {
+            clusterMember.setClusterMemberIp("127.0.0.1");
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java]: Same issue as above.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java (hunk 2 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java]: Difference between DB model and API model. We need to discuss it. For now, that code is there not to break service

File: src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java (hunk 3 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java]: Difference between DB model and API model. We need to discuss it. For now, that code is there not to break service

**Comment by dejan-brkic** [comment]
Aligned DB model with API model

File: src/main/java/org/craftercms/studio/api/v2/dal/ClusterDAO.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+package org.craftercms.studio.api.v2.dal;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ClusterDAO {
+
+    List<ClusterMember> getAllMembers();
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v2/dal/ClusterDAO.java:31]: JavaDoc please

**Review by sumerjabri — CHANGES_REQUESTED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**